src/cdogs/pickup.c

```diff
@@ -356,6 +356,16 @@ static bool TryPickupGun(
 			: StrWeaponClass(AmmoGetById(&gAmmo, pe->u.Ammo.Id)->DefaultGun);
 	const int actorsGunIdx = ActorFindGun(a, wc);
 
+	// Don't pick up guns in the melee slot or melee weapons in a gun slot
+	if (a->gunIndex == MELEE_SLOT && wc->Type != GUNTYPE_MELEE)
```

Comment:
there's a third type of weapon - grenades, will need to avoid switching to grenades, since you can never have that slot selected, you can only throw grenades using the grenade button